lib/ansible/modules/cloud/google/gce_instance_template.py

```diff
@@ -0,0 +1,565 @@
+  preemptible:
+    description:
+      - Defines whether the instance is preemptible.
+    default: null
+  tags:
+    description:
+      - a comma-separated list of tags to associate with the instance
+    default: null
+  metadata:
+    description:
+      - a hash/dictionary of custom data for the instance;
+        '{"key":"value", ...}'
+    default: null
+  description:
+    description:
+      - description of instance template
+    default: null
+  disks:
+    description:
+      - a list of persistent disks to attach to the instance; a string value
+        gives the name of the disk; alternatively, a dictionary value can
+        define 'name' and 'mode' ('READ_ONLY' or 'READ_WRITE'). The first entry
+        will be the boot disk (which must be READ_WRITE).
+    default: null
+  nic_gce_struct:
+    description:
+      - Support passing in the GCE-specific
+        formatted networkInterfaces[] structure.
+    default: null
+  project_id:
+    description:
+      - your GCE project ID
+    default: null
+  pem_file:
+    description:
+      - path to the pem file associated with the service account email
+        This option is deprecated. Use 'credentials_file'.
+    default: null
+  credentials_file:
+    description:
+      - path to the JSON file associated with the service account email
+    default: null
+requirements:
+    - "python >= 2.6"
+    - "apache-libcloud >= 0.13.3, >= 0.17.0 if using JSON credentials,
+      >= 0.20.0 if using preemptible option"
+notes:
+  - JSON credentials strongly preferred.
+author: "Gwenael Pellen (@GwenaelPellenArkeup) <gwenael.pellen@arkeup.com>"
+'''
+
+EXAMPLES = '''
+# Usage
+- name: create instance template named foo
+  gce_instance_template:
+    name: foo
+    size: n1-standard-1
+    image_family: ubuntu-1604-lts
+    state: present
+    project_id: "your-project-name"
+    credentials_file: "/path/to/your-key.json"
+    service_account_email: "your-sa@your-project-name.iam.gserviceaccount.com"
+
+# Example Playbook
+- name: Compute Engine Instance Template Examples
+  hosts: localhost
+  vars:
+    service_account_email: "your-sa@your-project-name.iam.gserviceaccount.com"
+    credentials_file: "/path/to/your-key.json"
+    project_id: "your-project-name"
+  tasks:
+    - name: create instance template
+      gce_instance_template:
+        name: my-test-instance-template
+        size: n1-standard-1
+        image_family: ubuntu-1604-lts
+        state: present
+        project_id: "{{ project_id }}"
+        credentials_file: "{{ credentials_file }}"
+        service_account_email: "{{ service_account_email }}"
+    - name: delete instance template
+      gce_instance_template:
+        name: my-test-instance-template
+        size: n1-standard-1
+        image_family: ubuntu-1604-lts
+        state: absent
+        project_id: "{{ project_id }}"
+        credentials_file: "{{ credentials_file }}"
+        service_account_email: "{{ service_account_email }}"
+'''
+
+RETURN = '''
```

Comment:
Is empty because no new return variable returned.
Only standard return variable used.